internal/k6runner/k6runner.go

```diff
@@ -111,7 +112,8 @@ func (r Script) Run(ctx context.Context, registry *prometheus.Registry, logger l
 		return false, err
 	}
 
-	return !resultCollector.failure, nil
+	success := result.Error == "" && result.ErrorCode == ""
+	return success, nil
```

Comment:
Not sure if at this point we should build a new error with `result.Error` 🤔

Reply:
It makes sense to do I think

Reply:
Delved a bit more on what this error is used for, and it seems like it is just `log.Warn`ed?

https://github.com/grafana/synthetic-monitoring-agent/blob/e3403fb23b923802e4c0552c6a3da180ece97ab5/internal/prober/scripted/scripted.go#L57-L65

If this is the case we might want to return an error when the code is something that is not-on-us.

Reply:
Refined (slang for "made more complicated") the error handling logic in the latest commit, LMK what you think :)